src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -240,119 +243,160 @@ private CookieOptions BuildCookieOptions()
                 throw new ArgumentNullException(nameof(user));
             }
 
-            properties = properties ?? new AuthenticationProperties();
+            if (_inSignIn)
```

Comment:
???? 😕 this smells terrible.
Did you repro this error? How?

Reply:
Yeah without this guard its trivial to hit, all it takes is a virtual scheme loop of any sort we have indirection now...  so 

Scheme A => Scheme B => Scheme A...  I added the alias to every test which is the minimal cycle.

Reply:
And yes I totally agree its kinda gross

Reply:
Will stack look informative when they get this exception?

Reply:
Its not too bad:
```
Authenticate for scheme:[Google] resulted in a recursive call back to itself.
   at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.<AuthenticateAsync>d__51.MoveNext() in C:\Github\Security\src\Microsoft.AspNetCore.Authentication\AuthenticationHandler.cs:line 141
   at Microsoft.AspNetCore.Authentication.AuthenticationService.<AuthenticateAsync>d__10.MoveNext()
   at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.<AuthenticateAsync>d__51.MoveNext() in C:\Github\Security\src\Microsoft.AspNetCore.Authentication\AuthenticationHandler.cs:line 147
   at Microsoft.AspNetCore.Authentication.AuthenticationService.<AuthenticateAsync>d__10.MoveNext()
   at Microsoft.AspNetCore.Authentication.AuthenticationHandler`1.<AuthenticateAsync>d__51.MoveNext() in C:\Github\Security\src\Microsoft.AspNetCore.Authentication\AuthenticationHandler.cs:line 147
   at Microsoft.AspNetCore.Authentication.AuthenticationService.<AuthenticateAsync>d__10.MoveNext()
   at Xunit.Assert.<RecordExceptionAsync>d__79.MoveNext() in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\Record.cs:line 82
```

Reply:
I'll update the error messages to talk about the appropriate option, so something more explicit like:

"Authenticate for scheme:[Google] resulted in a recursive call back to itself. Check for cycles in either ForwardAuthenticate, ForwardDefault, or ForwardDefaultSelector."